src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.TestClasses.cs

```diff
@@ -129,7 +129,7 @@ public class ClassWhereParametersHaveDefaultValue
             public StringComparison? NSC { get; }
             public char? NC { get; }
 
-            public ClassWhereParametersHaveDefaultValue(string? name, string address, 
+            public ClassWhereParametersHaveDefaultValue(string? name = "John Doe", string address = "1 Microsoft Way",
```

Comment:
>ClassWhereParametersHaveDefaultValue [](http://example.com/codeflow?start=19&length=36)

do we have any test using [Primary constructor](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/primary-constructors) with default values?

Reply:
We could add a test for that.  From [metadata I think it's the same](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKAPQGYACLUgYQBsIBnOgdQAsBTAJ1YAUJ2IxWAFw50AEhABurACKsAZhACu1QQDUI1RawAU6bAAYA/KRj9WpALykARACkA9sxilp91tbjkDpCAgScGSxtsUgBZKABjdns6ezlBUkYIAE8PAlIMzIzYBIgAc3MrABZMTzlqewgEuSCSgDp9OU8Ee0VganMEIOI67CLsAFYATmbWCKhIalIwbt7+4cwANgHiAYAOIYB2IeJMYjWixaWi4mJQzwBlQXZYPMp7MAAHXihY5zoIoKubmDuH55ubzqAHl2AhYBpPBFmLxSJ8rAByACOCLg6SymRyxlMBVqpVI5UqgmxNWKmAaTVILTaHWxXSsPT6gxGVLGEw02JmDLmzKWK3WWx2ewOR0WJzOl2ut3uTxeb2xHy+Ut+MoBr3sMBBYIh1ChMPY2PhpGRCIAlAQAN7ojIkLxGUgAOTMpAtpAKggA3KQAL5BUwCD3W8hkPT6UgAQT8AToLrdQi9vqsvn8rAYgfwWVtOQjuNd7oTQXyrHTmbIhKqpAAYrH8z6gnIS5lbdT2uZpDX43WrAhGzayAg2ZMwh3PV3pr3g6RvtL/nKNVPKCOC1YPhPbdDYYu853E3C12QsY7w7m46Pdzji0HbeXiY7q9uz36G1f+61W8YHe2H8uTD2X6zxkmD9QiXMcYDAfcp2VP5ZUBDUPwuLdTx/GBV3/DcDUdJDa3PCJ029AgCCAA=), but it's possible the roslyn representation seen by source gen is different.

Reply:
From my experience primary constructors are represented the same way in the symbols API, but it can't hurt to add a few tests.